Editor/Scripts/CodeGenerators/GenerateRGStateClasses.cs

```diff
@@ -97,12 +103,13 @@ public static void Generate(List<RGStateAttributesInfo> rgStateAttributesInfos)
                                     GenerateStateEntityFields(rgStateAttributeInfo.State)
                                 ),
                             classDeclaration
-
                         );
 
                     // Add the namespace declaration to the compilation unit
                     compilationUnit = compilationUnit.AddMembers(namespaceDeclaration);
 
+                    compilationUnit = compilationUnit.AddMembers(CodeGeneratorUtils.CreatePartialAttacherClass(rgStateAttributeInfo.NameSpace, componentType, $"RegressionGames.RGBotConfigs.{className}"));
```

Comment:
I think we should consider this having a `RequireComponent` for the behavior we found the RGState or RGAction attribute in as well.  Thus if they remove that behavior they will get warned that our scripts require it.